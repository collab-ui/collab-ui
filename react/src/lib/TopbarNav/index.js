/** @component topbar */

import React from 'react';
import PropTypes from 'prop-types';
import { List } from '@momentum-ui/react';
import { prefix } from '../utils/index';

class TopbarNav extends React.Component {
  render() {
    const {
      children,
      className,
      ...otherProps
    } = this.props;

    return (
      <nav
        className={
          `${prefix}-top-bar__nav` +
          `${className && ` ${className}` || ''}`
        }
        {...otherProps}
      >
        <List tabType='horizontal'>{children}</List>
      </nav>
    );
  }
}

TopbarNav.propTypes = {
  /** @prop Children node to render inside of TopbarNav | null */
  children: PropTypes.node,
  /** @prop Optional CSS class string | '' */
  className: PropTypes.string,
};

TopbarNav.defaultProps = {
  children: null,
  className: '',
};

TopbarNav.displayName = 'TopbarNav';

export default TopbarNav;
