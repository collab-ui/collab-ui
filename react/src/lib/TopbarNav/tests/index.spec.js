import React from 'react';
import { shallow } from 'enzyme';
import TopbarNav from '../TopbarNav';

describe('tests for <TopbarNav />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<TopbarNav id="test" />);

    expect(container).toMatchSnapshot();
  });

  it('should render one TopbarNav', () => {
    const container = shallow(<TopbarNav />);

    expect(container.find('.cui-top-bar__nav').length).toEqual(1);
  });

  it('should render children', () => {
    const container = shallow(
      <TopbarNav>
        <div className="testingforTbN" />
      </TopbarNav>
    );

    expect(container.find('.testingforTbN').length).toEqual(1);
  });
});
