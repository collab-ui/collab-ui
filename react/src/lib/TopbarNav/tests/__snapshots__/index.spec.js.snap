// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <TopbarNav /> should match SnapShot 1`] = `
<nav
  className="md-top-bar__nav"
  id="test"
>
  <List
    active={null}
    ariaConfig={null}
    className=""
    focus={null}
    focusFirst={true}
    focusFirstQuery=""
    focusQuery=""
    id={null}
    itemRole="listitem"
    navigationDirection="both"
    onSelect={null}
    role="list"
    shouldFocusActive={false}
    shouldFocusInitial={true}
    shouldLoop={true}
    shouldPropagateKeyDown={false}
    tabType="horizontal"
    trackActive={true}
    type={null}
    wrap={false}
  />
</nav>
`;
